pkg/endpoint/events.go

```diff
@@ -119,6 +120,84 @@ func (e *Endpoint) PolicyRevisionBumpEvent(rev uint64) {
 	}
 }
 
+/// EndpointNoTrackEvent contains all fields necessary to update the NOTRACK rules.
+type EndpointNoTrackEvent struct {
+	ep     *Endpoint
+	annoCB AnnotationsResolverCB
+}
+
+// Handle handles the NOTRACK rule update.
+func (ev *EndpointNoTrackEvent) Handle(res chan interface{}) {
+	var oldPort uint16
+
+	e := ev.ep
+
+	// If this endpoint is going away, remove all related NOTRACK rules.
+	if err := e.lockAlive(); err != nil {
+		res <- &EndpointRegenerationResult{
+			err: nil,
+		}
+		return
+	}
+
+	defer e.unlock()
+
+	port, err := ev.annoCB(e.K8sNamespace, e.K8sPodName)
+	if err != nil {
+		res <- &EndpointRegenerationResult{
+			err: err,
+		}
+		return
+	}
+
+	// Validate annotation before we do any actual alteration to the endpoint.
+	p64, err := strconv.ParseUint(port, 10, 16)
+	// Port should be within [1-65535].
+	if err != nil || p64 == 0 || p64 > 65535 {
```

Comment:
The value cannot be > 65535 if ParseUint() takes bitsize 16 as the third parameter:

https://play.golang.org/p/RlYnjnXPcsx

Reply:
makes sense, done.